src/Orleans.Transactions/State/StorageBatch.cs

```diff
@@ -256,5 +257,24 @@ public void FollowUpAction(Action action)
             }
             followUpActions.Add(action);
         }
+
+        public void AddStorePreCondition(Func<Task<bool>> action)
+        {
+            if (this.storeConditions == null)
```

Comment:
why do we lazy init this one? can we init this in the constructor?

Reply:
We can.  I used lazy init cause:
- Only toc uses it, so it won't be used most of the time.
- Was following the pattern used by the after actions (the 'when in rome' rule).

I don't mind changing this if you feel strongly that it should be initialized in the constructor.

Reply:
It is a small fix, I'd like for it to be fixed here. the more cleaner code is, the better for maintenance

Reply:
Will do, sorry I didn't get back to the feedback in time.. :/